### PR TITLE
Add hex6C rev2 PCB to QMK

### PR DESCRIPTION
**Pull request by mechlovin**
## Description

Add Hex6C Rev. 2 PCB to QMK
Thank you.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mechlovin/hex6c/config.h [hunk 1]
```diff
@@ -19,39 +19,16 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #include "config_common.h"
 
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x4D4C
+#define PRODUCT_ID   0x6C01
+#define DEVICE_VER   0x0001
+#define MANUFACTURER Mechlovin Studio and Hex Keyboard
+#define PRODUCT      Hex6C
+
```
Comment (drashna) [inline keyboards/mechlovin/hex6c/config.h:28]: Please revert this.  This information should be located in the info.json.

File: keyboards/mechlovin/hex6c/info.json [hunk 2]
```diff
@@ -1,13 +1,7 @@
...
-    "keyboard_name": "Hex6C",
-    "manufacturer": "Mechlovin Studio and Hex Keyboard",
+    "keyboard_name": "hex-6C",
     "url": "",
     "maintainer": "Mechlovin' Studio",
-    "usb": {
-        "vid": "0x4D4C",
-        "pid": "0x6C01",
-        "device_version": "0.0.1"
-    },
```
Comment (drashna) [inline keyboards/mechlovin/hex6c/info.json:10]: please revert

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/mechlovin/hex6c/rev2/readme.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+Make example for this keyboard (after setting up your build environment):
+
+    make mechlovin/hex6c/rev2:default
+    make mechlovin/hex6c/rev2:via
+
+Flashing example for this keyboard:
+
+    make mechlovin/hex6c/rev2:default:flash
+    make mechlovin/hex6c/rev2:via:flash
```
Comment (fauxpark) [inline keyboards/mechlovin/hex6c/rev2/readme.md:19]:
```suggestion

Flashing example for this keyboard:

    make mechlovin/hex6c/rev2:default:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by drashna** [comment]
This has a bunch of merge conflicts, that needs to be resolved.

**Pull request closed without merging**